### PR TITLE
Add integration test for binary vector values

### PR DESCRIPTION
**Pull request by VijayanB**
### Description
Add integration test for binary vector values

### Related Issues
Part of #1942 

### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md).
- [x] Commits are signed per the DCO using `--signoff`.
- [ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/k-NN/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/java/org/opensearch/knn/index/codec/KNN80Codec/KNN80DocValuesProducer.java [hunk 1]
```diff
@@ -75,6 +75,9 @@ public KNN80DocValuesProducer(DocValuesProducer delegate, SegmentReadState state
                 continue;
             }
             List<String> engineFiles = KNNCodecUtil.getEngineFiles(knnEngine.getExtension(), field.name, state.segmentInfo);
+            if (engineFiles.isEmpty()) {
```
Comment (jmazanec15) [inline src/main/java/org/opensearch/knn/index/codec/KNN80Codec/KNN80DocValuesProducer.java:78]: Can we change description to account for this fix?

**Review by jmazanec15 — APPROVED** [review]
LGTM - please update the description though because were changing prod code - not just tests

File: src/main/java/org/opensearch/knn/index/codec/KNN80Codec/KNN80DocValuesProducer.java (hunk 1 above)
Comment (VijayanB) [reply src/main/java/org/opensearch/knn/index/codec/KNN80Codec/KNN80DocValuesProducer.java:78]: Created #2147 to address this change in main. I will rebase once it is merged.

**Comment by VijayanB** [comment]
CI failure with flaky test https://github.com/opensearch-project/k-NN/issues/2169

**Review by navneet1v — APPROVED** [review]

**Pull request merged**